src/mame/heathkit/z37_fdc.cpp

```diff
@@ -170,6 +170,16 @@ void heath_z37_fdc_device::device_start()
 	m_access_track_sector = false;
 }
 
+void heath_z37_fdc_device::device_reset()
+{
+	m_control_reg = 0;
+	m_interface_reg = 0;
+
+	m_intrq_allowed = false;
+	m_drq_allowed = false;
+	m_access_track_sector = false;
+}
```

Comment:
If the interrupt and/or DMA request outputs were asserted before the device was reset, they should be deasserted due to `m_intrq_allowed` and `m_drq_allowed` becoming false here, shouldn’t they?  You need to call the callbacks to notify whatever’s connected of the change.